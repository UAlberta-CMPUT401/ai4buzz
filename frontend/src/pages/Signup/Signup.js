import React, { useState } from 'react';
import Cookies from 'js-cookie';

import { signup } from '../../services/auth';
import SigninForm from '../../components/SigninForm/SigninForm';
import styles from './Signup.module.css';
import LoadingSpinner from '../../components/LoadingOverlay/LoadingOverlay';

const Signup = () => {
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();

  const handleSubmit = async () => {
    const res = await signup(email, password);
    Cookies.set('access_token', res.data.access_token, { path: '/' });
  };

  return (
    <div className={styles.signupPage}>
      <h1 className={styles.title}>AI4Buzz</h1>
      <SigninForm
        authType="Signup"
        onEmailChange={setEmail}
        onPasswordChange={setPassword}
        onSubmit={handleSubmit}
      />
    </div>
  );
};

export default Signup;
