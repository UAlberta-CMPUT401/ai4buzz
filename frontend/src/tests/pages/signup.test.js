import React from 'react';
import renderer from 'react-test-renderer';
import Signup from '../../pages/Signup/Signup';
import { BrowserRouter } from 'react-router-dom';

it('Renders signup page correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Signup />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
