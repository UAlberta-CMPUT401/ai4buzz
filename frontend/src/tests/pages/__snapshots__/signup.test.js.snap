// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders signup page correctly 1`] = `
<div
  className="signupPage"
>
  <h1
    className="title"
  >
    AI4Buzz
  </h1>
  <form
    className="signinForm"
    onSubmit={[Function]}
  >
    <h1>
      Signup
    </h1>
    <label
      className="labelName"
    >
      Email
    </label>
    <input
      className="signinInputEmail"
      id="signin__email__input"
      onChange={[Function]}
      placeholder="email"
      type="email"
    />
    <label
      className="labelName"
    >
      Password
    </label>
    <input
      className="signinInputPassword"
      id="signin__password__input"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
    <div
      className="buttonGradient"
    >
      <button
        className="submitButton"
      >
        Signup
      </button>
    </div>
    <a
      className="switchAuth"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </form>
</div>
`;
